src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -0,0 +1,337 @@
+        {
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        internal override int CalculateLocalSyntaxOffset(int localPosition, SyntaxTree localTree)
+        {
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        internal override bool TryGetThisParameter(out ParameterSymbol thisParameter)
+        {
+            // Local function symbols have no "this" parameter
+            thisParameter = null;
+            return true;
+        }
+
+        private ImmutableArray<TypeParameterSymbol> MakeTypeParameters(DiagnosticBag diagnostics)
+        {
+            var result = ArrayBuilder<TypeParameterSymbol>.GetInstance();
+            var typeParameters = _syntax.TypeParameterList.Parameters;
+            for (int ordinal = 0; ordinal < typeParameters.Count; ordinal++)
+            {
+                var parameter = typeParameters[ordinal];
+                var identifier = parameter.Identifier;
+                var location = identifier.GetLocation();
+                var name = identifier.ValueText;
+
+                // TODO: Add diagnostic checks for nested local functions (and containing method)
+                if (name == this.Name)
+                {
+                    diagnostics.Add(ErrorCode.ERR_TypeVariableSameAsParent, location, name);
+                }
+
+                for (int i = 0; i < result.Count; i++)
+                {
+                    if (name == result[i].Name)
+                    {
+                        diagnostics.Add(ErrorCode.ERR_DuplicateTypeParameter, location, name);
+                        break;
+                    }
+                }
+
+                var tpEnclosing = ContainingType.FindEnclosingTypeParameter(name);
+                if ((object)tpEnclosing != null)
+                {
+                    // Type parameter '{0}' has the same name as the type parameter from outer type '{1}'
+                    diagnostics.Add(ErrorCode.WRN_TypeParameterSameAsOuterTypeParameter, location, name, tpEnclosing.ContainingType);
+                }
+
+                var typeParameter = new LocalFunctionTypeParameterSymbol(
+                        this,
+                        name,
+                        ordinal,
+                        ImmutableArray.Create(location),
+                        ImmutableArray.Create(parameter.GetReference()));
+
+                result.Add(typeParameter);
+            }
+
+            return result.ToImmutableAndFree();
+        }
+
+        public override int GetHashCode()
+        {
+            // this is what lambdas do (do not use hashes of other fields)
+            return _syntax.GetHashCode();
+        }
+
+        public sealed override bool Equals(object symbol)
+        {
+            if ((object)this == symbol) return true;
+
+            var localFunction = symbol as LocalFunctionSymbol;
+            return (object)localFunction != null
+                && localFunction._syntax == _syntax
```

Comment:
Once you've found it is the same syntax, do you really need to check anything else?